specifyweb/frontend/js_src/lib/components/AttachmentsBulkImport/ViewAttachmentFiles.tsx

```diff
@@ -0,0 +1,214 @@
+                <div>
+                  {keyLocalizationMapAttachment[resolvedRecord.reason] ?? ''}
+                </div>
+              )
+            )}
+          </div>,
+        ],
+
+        attachmentId,
+        canDisambiguate: typeof handleDisambiguate === 'function',
+        isNativeError: resolvedRecord?.type === 'invalid',
+        isRuntimeError,
+      } as const;
+    }
+  );
+
+export function ViewAttachmentFiles({
+  uploadableFiles,
+  baseTableName,
+  uploadSpec,
+  onDisambiguation: handleDisambiguation,
+  onFilesDropped: handleFilesDropped,
+}: {
+  readonly uploadableFiles: RA<PartialUploadableFileSpec>;
+  readonly baseTableName: keyof Tables | undefined;
+  readonly uploadSpec: PartialAttachmentUploadSpec;
+  readonly onDisambiguation:
+    | ((
+        disambiguatedId: number,
+        indexToDisambiguate: number,
+        multiple: boolean
+      ) => void)
+    | undefined;
+  readonly onFilesDropped?: (file: FileList) => void;
+}): JSX.Element | null {
+  const [disambiguationIndex, setDisambiguationIndex] = React.useState<
+    number | undefined
+  >(undefined);
+
+  const data = React.useMemo(
+    () =>
+      resolveAttachmentDatasetData(
+        uploadableFiles,
+        setDisambiguationIndex,
+        baseTableName
+      ),
+    [uploadableFiles, setDisambiguationIndex, baseTableName]
+  );
+  const headers = React.useMemo(
+    () => ({
+      selectedFileName: commonText.selectedFileName(),
+      fileSize: attachmentsText.fileSize(),
+      record: (
+        <div className="flex min-w-fit items-center gap-2">
+          {baseTableName === undefined ? (
+            formsText.record()
+          ) : (
+            <>
+              <TableIcon label name={baseTableName} />
+              {uploadSpec.staticPathKey === undefined
+                ? ''
+                : defined(
+                    strictGetModel(baseTableName).getField(
+                      staticAttachmentImportPaths[uploadSpec.staticPathKey].path
+                    )
+                  ).label}
+            </>
+          )}
+        </div>
+      ),
+      status: attachmentsText.status(),
+      attachmentId: attachmentsText.attachmentID(),
+    }),
+    [uploadSpec.staticPathKey]
+  );
+
+  const fileDropDivRef = React.useRef<HTMLDivElement>(null);
+  const { isDragging, ...restCallbacks } = useDragDropFiles(
+    handleFilesDropped,
+    fileDropDivRef
+  );
+
+  const headerElements = React.useMemo(
+    () => Object.values(headers).map((label) => <>{label}</>),
+    [headers]
+  );
+
+  return data.length === 0 ? null : (
+    <>
+      <div
+        className="flex w-full flex-1 flex-col gap-2 overflow-auto rounded bg-[color:var(--background)] p-4 shadow-md"
+        {...restCallbacks}
+      >
+        <div className="font-semibold">
+          {attachmentsText.totalFiles({ fileCount: data.length })}
+        </div>
+        <div className="h-full overflow-auto" ref={fileDropDivRef}>
+          <GenericSortedDataViewer
```

Comment:
should there be any prevention against uploading duplicate files? i.e if file name is the same and file size is the same? ignore duplicates or flag them?

Reply:
**Conclusion:**
Flag and skip them

Reply:
fixed